packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -125,10 +125,7 @@ class IOSDevice extends Device {
           ephemeral: true,
       ) {
     if (!platform.isMacOS) {
-      printError('Cannot control iOS devices or simulators. ideviceinstaller and iproxy are not available on your platform.');
-      _installerPath = null;
-      _iproxyPath = null;
-      return;
+      throw UnsupportedError('Control of iOS devices or simulators only supported on Mac OS.');
```

Comment:
Sorry for only now thinking about this a bit more, but I wonder if we should instead do the following:
1. Initialize `_installerPath` and `_iproxyPath` only on the first access.
2. throwing the UnsupportedError at the very top of getAttachedDevices().

Reply:
I can write getters that only initialize the variables when called. Curious, what's the advantage of moving the platform check later? To allow instantiating the class on non mac platforms?